src/sputils/doesMsgHaveError.js

```diff
@@ -14,7 +14,7 @@ define(["jquery"], function($){
             spErrCode   = $msg.find("ErrorCode"),
             response    = false;
 
-        if ( !spErrCode.length ) {
+        if ( !spErrCode.length || spErrCode.text() === "NoError") {
```

Comment:
I think the most appropriate place for checking this condition is further below on line 31 where this method actually checks the content of the <error> tag...
This block of code is only checking if we should continue to look for a XML tag that holds the request's status... in this case, we already found it, so there is no need to keep looking (it's unlikely (I think :) ) that the response would have both an <error> and <faultcode> tags).